pkg/tlsutil/config.go

```diff
@@ -0,0 +1,97 @@
+// Package tlsutil provides primitives to retrieve secure-enough TLS configurations for both clients and servers.
+//
+// As a reminder from https://golang.org/pkg/crypto/tls/#Config:
+//	A Config structure is used to configure a TLS client or server. After one has been passed to a TLS function it must not be modified.
+//	A Config may be reused; the tls package will also not modify it.
+package tlsutil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// NewConfig returns a secure-enough-by-default TLS configuration object.
+func NewConfig() *tls.Config {
+	return &tls.Config{
+		// Avoid fallback to SSL protocols < TLS1.0
+		MinVersion:               tls.VersionTLS10,
+		PreferServerCipherSuites: true,
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+		},
+	}
+}
+
+// certPool returns an X.509 certificate pool from `ca`, the certificate file.
+func certPool(ca string) (*x509.CertPool, error) {
+	// If we should verify the server, we need to load a trusted ca
+	certPool := x509.NewCertPool()
+	file, err := ioutil.ReadFile(ca)
+	if err != nil {
+		return nil, fmt.Errorf("Could not read CA certificate %s: %v", ca, err)
+	}
+	certPool.AppendCertsFromPEM(file)
+	return certPool, nil
+}
+
+// ClientConfig returns a TLS configuration meant to be used by a client.
+func ClientConfig(enable, verify bool, ca, cert, key string) (*tls.Config, error) {
+	if !enable {
+		return nil, nil
+	}
+	tlsConfig := NewConfig()
+	if verify {
+		tlsConfig.InsecureSkipVerify = false
+		certPool, err := certPool(ca)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.RootCAs = certPool
+	} else {
+		tlsConfig.InsecureSkipVerify = true
+	}
+
+	if cert != "" && key != "" {
```

Comment:
This is a logic change, previously the cert and key would be stat'ed and any error would cause it to skip. I don't necessarily agree with that logic but it should be called out because now it will error if a file does not exist.